pkg/apis/policy/validation/validation.go

```diff
@@ -476,6 +477,35 @@ func validatePSPCapsAgainstDrops(requiredDrops []core.Capability, capsToCheck []
 	return allErrs
 }
 
+// validateRuntimeClassStrategy ensures all the RuntimeClass restrictions are valid.
+func validateRuntimeClassStrategy(fldPath *field.Path, rc *policy.RuntimeClassStrategyOptions) field.ErrorList {
+	if rc == nil {
+		return nil
+	}
+
+	var allErrs field.ErrorList
+
+	if rc.DefaultRuntimeClassName != nil {
+		allErrs = append(allErrs, apivalidation.ValidateRuntimeClassName(*rc.DefaultRuntimeClassName, fldPath.Child("defaultRuntimeClassName"))...)
+	}
+
+	defaultAllowed := rc.DefaultRuntimeClassName == nil || rc.AllowedRuntimeClassNames == nil // Ignore this if the default is nil
```

Comment:
change to make "*" an explicit "allow all"

Reply:
and in validation, if the allowed names list contains "*", it should be the only item in the list